pkg/backend/package_registry.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2025, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backend
+
+import (
+	ctx "context"
+	"io"
+
+	"github.com/blang/semver"
+)
+
+type PackagePublishOp struct {
```

Comment:
This seems to be the exact same type as `PublishPackageInput` do we really need both?

Reply:
Yeah, this would create circular dependencies otherwise :(

Reply:
Factoring it out into a separate package wouldn't work either?

Reply:
We could put it in a separate package, yes that'd work. At the same time, I think having the separation between the client and the backend is actually helpful in order to be able to evolve them separately. They're the same now, but going forward I can see them diverge. I'm not certain what the more maintainable option is, so I'm happy to go either way

Reply:
It seems to me that they would always be pretty much the same. I feel like we have a lot of accidental duplication in `pulumi/pulumi` already, so I'd prefer if they were merged, but I'm not going to be too upset if you disagree and they stay as they are :)

Reply:
I ended up consolidating the two. Thanks for the inputs! :)